// components/login/index.js
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link, Redirect } from 'react-router-dom'
import PropTypes from 'prop-types'
import { browserHistory, withRouter } from 'react-router'

import { postLogin } from 'actions'

class UserLogin extends Component {
    constructor (props) {
        super(props)

        this.state = {
        login: '',
        password: '',
        error: '',
        isLoading: false,
        isLoggingIn: false,
        notification: false,
        logged: false
        }
    }

    onSubmit (event) {
        event.preventDefault()
        let input = {}
        input.login = this.state.login
        input.password = this.state.password
        if (input.login !== '' && input.password !== '') {
        this.setState({isLoggingIn: true, isLoading: true})
        this.props.postLogin(input).then((response) => {
            if (response.success) {
                this.setState({
                    isLoading: false,
                    isLoggingIn: false,
                    notification: true,
                    login: '',
                    password: '',
                    error: ''
            })
            this.setState({logged: true})
            } else {
            this.setState({
                isLoading: false,
                isLoggingIn: false,
                error: response.errors[0].message,
                notification: false
            })
            }
        })
        } else {
        this.setState({
            error: 'Введите корректные данные',
            notification: false
        })
        }
    }

    onChange (event) {
        this.setState({
        [event.target.name]: event.target.value
        })
    }

    render () {
        return (
        <div className="register">
            <h2 className="register__name">Вход</h2>
            <br/>
            {this.state.error ? <p className="register-form__message">{this.state.error}</p>: ''}
            {this.state.message ? <p className="register-form__message">{this.state.message}</p>: ''}

            <form className="register-form" onSubmit={this.onSubmit.bind(this)}>  
                <div className="register-form__row">
                    <img className="register-form__icon" src="/icons/user.png" alt=""/> 
                    <div className="register-form__inputArea">
                        <input 
                        className="register-form__input" 
                        name="login"
                        value={this.state.login}
                        placeholder="Логин"
                        onChange={this.onChange.bind(this)}
                        />
                    </div>
                </div>
                <div className="register-form__row">
                    <img className="register-form__icon" src="/icons/password.png" alt=""/> 
                    <div className="register-form__inputArea">
                        <input 
                        className="register-form__input" 
                        type="password"
                        name="password"
                        value={this.state.password}
                        placeholder="Пароль"
                        onChange={this.onChange.bind(this)}
                        />
                    </div>
                </div>
                
                <input className="basket__btn" type="submit" value="Войти" />
            </form>
            {this.state.logged ?  browserHistory.push('/') : ''}
        </div>
        )
    }
}

UserLogin.propTypes = {
  postLogin: PropTypes.func.isRequired
}

UserLogin.contextTypes = {
  router: PropTypes.object.isRequired
}

export default connect(null, {postLogin})(UserLogin)
