// api/mockComics.js
export default [
    {
        id: "1",
        categoryId: "1",
        tag: "Fresh",
        year: "2016",
        size: "7,48 x 11,02 inch",
        sheets: "392",
        image: "/uploads/Superman.jpg",
        price: "23",
        publish: "DC",
        name: "Marvel",
        description: "Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis ea eos, consectetur repudiandae ad vitae reiciendis et? Iusto dicta ducimus, rem omnis molestias ullam reprehenderit alias. Magni modi temporibus maiores, a quos dolore, numquam nam cumque non ea, facilis libero facere impedit nulla. Aliquam accusantium aliquid ut. Corporis, repudiandae inventore."
    },
    {
        id: "2",
        categoryId: "2",
        tag: "Fresh",
        year: "2016",
        size: "7,48 x 11,02 inch",
        sheets: "392",
        image: "/uploads/Superman.jpg",
        price: "14",
        publish: "DC",
        name: "Supermen Unchained #1",
        description: "Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis ea eos, consectetur repudiandae ad vitae reiciendis et? Iusto dicta ducimus, rem omnis molestias ullam reprehenderit alias. Magni modi temporibus maiores, a quos dolore, numquam nam cumque non ea, facilis libero facere impedit nulla. Aliquam accusantium aliquid ut. Corporis, repudiandae inventore."
    },
    {
        id: "3",
        categoryId: "2",
        tag: "Fresh",
        year: "2016",
        size: "7,48 x 11,02 inch",
        sheets: "392",
        image: "/uploads/Superman.jpg",
        price: "16",
        publish: "DC",
        name: "Marvel",
        description: "Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis ea eos, consectetur repudiandae ad vitae reiciendis et? Iusto dicta ducimus, rem omnis molestias ullam reprehenderit alias. Magni modi temporibus maiores, a quos dolore, numquam nam cumque non ea, facilis libero facere impedit nulla. Aliquam accusantium aliquid ut. Corporis, repudiandae inventore."
    },
    {
        id: "4",
        categoryId: "1",
        tag: "Fresh",
        year: "2016",
        size: "7,48 x 11,02 inch",
        sheets: "392",
        image: "/uploads/Superman.jpg",
        price: "14",
        publish: "DC",
        name: "Marvel",
        description: "Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis ea eos, consectetur repudiandae ad vitae reiciendis et? Iusto dicta ducimus, rem omnis molestias ullam reprehenderit alias. Magni modi temporibus maiores, a quos dolore, numquam nam cumque non ea, facilis libero facere impedit nulla. Aliquam accusantium aliquid ut. Corporis, repudiandae inventore."
    },
    {
        id: "5",
        categoryId: "3",
        tag: "Fresh",
        year: "2016",
        size: "7,48 x 11,02 inch",
        sheets: "392",
        image: "/uploads/Superman.jpg",
        price: "17",
        publish: "DC",
        name: "Marvel",
        description: "Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis ea eos, consectetur repudiandae ad vitae reiciendis et? Iusto dicta ducimus, rem omnis molestias ullam reprehenderit alias. Magni modi temporibus maiores, a quos dolore, numquam nam cumque non ea, facilis libero facere impedit nulla. Aliquam accusantium aliquid ut. Corporis, repudiandae inventore."
    },
    {
        id: "6",
        categoryId: "1",
        tag: "Fresh",
        year: "2016",
        size: "7,48 x 11,02 inch",
        sheets: "392",
        image: "/uploads/Superman.jpg",
        price: "32",
        publish: "DC",
        name: "Marvel",
        description: "Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis ea eos, consectetur repudiandae ad vitae reiciendis et? Iusto dicta ducimus, rem omnis molestias ullam reprehenderit alias. Magni modi temporibus maiores, a quos dolore, numquam nam cumque non ea, facilis libero facere impedit nulla. Aliquam accusantium aliquid ut. Corporis, repudiandae inventore."
    },
    {
        id: "7",
        categoryId: "3",
        tag: "Fresh",
        year: "2016",
        size: "7,48 x 11,02 inch",
        sheets: "392",
        image: "/uploads/Superman.jpg",
        price: "14",
        publish: "DC",
        name: "Marvel",
        description: "Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis ea eos, consectetur repudiandae ad vitae reiciendis et? Iusto dicta ducimus, rem omnis molestias ullam reprehenderit alias. Magni modi temporibus maiores, a quos dolore, numquam nam cumque non ea, facilis libero facere impedit nulla. Aliquam accusantium aliquid ut. Corporis, repudiandae inventore."
    },
    {
        id: "8",
        categoryId: "1",
        tag: "Fresh",
        year: "2016",
        size: "7,48 x 11,02 inch",
        sheets: "392",
        image: "/uploads/Superman.jpg",
        price: "14",
        publish: "DC",
        name: "Marvel",
        description: "Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quis ea eos, consectetur repudiandae ad vitae reiciendis et? Iusto dicta ducimus, rem omnis molestias ullam reprehenderit alias. Magni modi temporibus maiores, a quos dolore, numquam nam cumque non ea, facilis libero facere impedit nulla. Aliquam accusantium aliquid ut. Corporis, repudiandae inventore."
    },
];
